src/QsCompiler/LanguageServer/FileSystemWatcher.cs

```diff
@@ -84,51 +90,62 @@ private System.IO.FileSystemWatcher GetWatcher(string folder, string pattern, No
+            }
             var root = Uri.TryCreate(path, UriKind.Absolute, out Uri uri) ? uri : null;
 
             var success = Directory.EnumerateDirectories(root.LocalPath).TryEnumerate(out var subfolders);
             success = globPatterns.TryEnumerate(
-                pattern => Directory.EnumerateFiles(root.LocalPath, pattern, SearchOption.TopDirectoryOnly), out var files) 
+                pattern => Directory.EnumerateFiles(root.LocalPath, pattern, SearchOption.TopDirectoryOnly), out var files)
                 && success;
 
             directories[root] = subfolders.Concat(files.SelectMany(items => items)).Select(Path.GetFileName).ToImmutableHashSet();
             foreach (var subfolder in subfolders)
-            { success = GlobDirectoryStructure(directories, subfolder, globPatterns) && success; }
+            {
+                success = GlobDirectoryStructure(directories, subfolder, globPatterns) && success;
+            }
 
             return success;
         }
 
         /// <summary>
-        /// Adds suitable listeners to capture all given glob patterns for the given folder, 
+        /// Adds suitable listeners to capture all given glob patterns for the given folder,
         /// and - if subfolders is set to true - all its subfolders.
-        /// Does nothing if no folder with the give path exists. 
+        /// Does nothing if no folder with the give path exists.
         /// </summary>
-        public Task ListenAsync(string folder, bool subfolders, Action<ImmutableDictionary<Uri, ImmutableHashSet<string>>> onInitialState, params string[] globPatterns) 
+        public Task ListenAsync(string folder, bool subfolders, Action<ImmutableDictionary<Uri, ImmutableHashSet<string>>> onInitialState, params string[] globPatterns)
         {
-            if (!Directory.Exists(folder)) return Task.CompletedTask;
+            if (!Directory.Exists(folder))
+            {
+                return Task.CompletedTask;
+            }
             folder = folder.TrimEnd(Path.DirectorySeparatorChar);
 
             globPatterns = globPatterns.Distinct().ToArray();
-            this.GlobPatterns.AddOrUpdate(new Uri(folder), globPatterns, (_, currentPatterns) => currentPatterns.Concat(globPatterns).Distinct().ToArray());
+            this.globPatterns.AddOrUpdate(new Uri(folder), globPatterns, (_, currentPatterns) => currentPatterns.Concat(globPatterns).Distinct().ToArray());
 
             var filters = globPatterns.Select(p => (p, NotifyFilters.FileName | NotifyFilters.LastWrite));
-            if (subfolders) filters = filters.Concat(new (string, NotifyFilters)[] { (String.Empty, NotifyFilters.DirectoryName) });
+            if (subfolders)
+            {
+                filters = filters.Concat(new (string, NotifyFilters)[] { (string.Empty, NotifyFilters.DirectoryName) });
+            }
 
-            return this.Processing.QueueForExecutionAsync(() =>
+            return this.processing.QueueForExecutionAsync(() =>
```

Comment:
This is an example of the style I prefer for lambdas.

Reply:
I think this style is allowed as long as there are no arguments after the lambda.

Reply:
Ah okay, I may try to order parameters for functions I write in the future appropriately, so that lambdas are last. That may be a good rule in general.